CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.16)
+
+# check if iceoryx is in CMAKE_PREFIX_PATH
+find_package(iceoryx_hoofs QUIET)
+find_package(iceoryx_posh QUIET)
+
+# fetch iceoryx if not found
+if(NOT iceoryx_hoofs_FOUND OR NOT iceoryx_posh_FOUND)
+    if(iceoryx_hoofs_FOUND)
+        message(FATAL_ERROR "iceoryx_hoofs was not found with 'find_package' but other parts were found!")
+    endif()
+    if(iceoryx_posh_FOUND)
+        message(FATAL_ERROR "iceoryx_posh was not found with 'find_package' but other parts were found!")
+    endif()
+
+    include(FetchContent)
+    FetchContent_Declare(
+        iceoryx
+        GIT_REPOSITORY https://github.com/eclipse-iceoryx/iceoryx.git
+        GIT_TAG master
+    )
+    FetchContent_GetProperties(iceoryx)
+    if (NOT iceoryx_POPULATED)
+        message(STATUS "updating: iceoryx" )
+        FetchContent_Populate(iceoryx)
+    endif()
+
+    set(ICEORYX_WITH_FETCH_CONTENT true CACHE INTERNAL "")
+    set(iceoryx_SOURCE_DIR ${iceoryx_SOURCE_DIR} CACHE INTERNAL "")
+    set(iceoryx_BINARY_DIR ${iceoryx_BINARY_DIR} CACHE INTERNAL "")
+endif()
+
+if(ICEORYX_WITH_FETCH_CONTENT)
+    add_subdirectory(${iceoryx_SOURCE_DIR}/iceoryx_hoofs ${iceoryx_BINARY_DIR}/iceoryx_hoofs)
+    add_subdirectory(${iceoryx_SOURCE_DIR}/iceoryx_posh  ${iceoryx_BINARY_DIR}/iceoryx_posh)
+
+    find_package(iceoryx_posh REQUIRED)
+    find_package(iceoryx_hoofs REQUIRED)
+endif()
+
+include(IceoryxPlatform)
+
+project(iceoryx_project_template)
+
+add_executable(publisher src/publisher.cpp)
```

Comment:
You could think about using `iox_add_executable` here. Reduces the CMake noise.

Reply:
I was thinking about using the v2.0.2 tag for this but if we want to track master we could already use `iox_add_executable`.
In general I'm more in favor of using the latest release tag. If someone wants to tinker with iceoryx it would be always a known working version.

@dkroenke @elfenpiff what do you think?

Reply:
I agree, let's use the latest release and add a hint that `v3.x` will provide the `iox_add_excutable` macro.

Reply:
I'll create an issue for this

Reply:
@elfenpiff then we would have to always check whether this template runs with the latest master. When it is pinned to a release, we only have to create an MR when we release iceoryx and merge after the CI ran so it is much less effort for us.

Reply:
We could leave it for now as it is. But the new idiomatic iceoryx way would be `iox_add_executable`. Could we maybe temporarily the current master hash and add a comment that we suggest tags?
But I would be fine with either way.